eva-server/src/test/java/uk/ac/ebi/eva/server/repository/VariantEntityRepositoryTest.java

```diff
@@ -0,0 +1,176 @@
+
+    @Autowired
+    private VariantEntityRepository variantEntityRepository;
+
+    @Test
+    public void testVariantIdIsFound(){
+        String id = "rs527639301";
+        final List<VariantEntity> variantEntityList = variantEntityRepository.findByIds(id);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() > 0);
+        Set<String> idSet = new HashSet<>();
+        idSet.add(id);
+        assertEquals(idSet, variantEntityList.get(0).getIds());
+    }
+
+    @Test
+    public void testNonExistentVariantIdIsNotFound(){
+        String id = "notarealid";
+        final List<VariantEntity> variantEntityList = variantEntityRepository.findByIds(id);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() == 0);
+    }
+
+    @Test
+    public void testVariantRegionIsFound(){
+        String chr = "20";
+        int start = 60343;
+        int end = 60343;
+        final List<VariantEntity> variantEntityList = variantEntityRepository.findByChrAndStartWithMarginAndEndWithMargin(chr, start, end);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() > 0);
+        assertEquals(chr, variantEntityList.get(0).getChromosome());
+        assertEquals(start, variantEntityList.get(0).getStart());
+        assertEquals(end, variantEntityList.get(0).getStart());
+    }
+
+    @Test
+    public void testVariantRegionIsFoundMultiple(){
+        String chr = "20";
+        int start = 60916;
+        int end = 61098;
+        final List<VariantEntity> variantEntityList = variantEntityRepository.findByChrAndStartWithMarginAndEndWithMargin(chr, start, end);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() > 0);
+        assertEquals(4, variantEntityList.size());
+        VariantEntity prevVariantEntity = variantEntityList.get(0);
+        for (VariantEntity currVariantEntity : variantEntityList) {
+            assertTrue(prevVariantEntity.getStart() <= currVariantEntity.getStart());
+        }
+    }
+
+    @Test
+    public void testNonExistentVariantRegionIsNotFound(){
+        String chr = "20";
+        int start = 61098;
+        int end = 60916;
+        final List<VariantEntity> variantEntityList = variantEntityRepository.findByChrAndStartWithMarginAndEndWithMargin(chr, start, end);
+        assertNotNull(variantEntityList);
+        assertTrue(variantEntityList.size() == 0);
+    }
+
+
+    @Configuration
+    @EnableMongoRepositories
+    @ComponentScan(basePackageClasses = { VariantEntityRepository.class })
+    static class VariantEntityRepositoryConfiguration extends AbstractMongoConfiguration {
+
+        @Autowired
+        private MongoDbFactory mongoDbFactory;
+
+        @Override
+        protected String getDatabaseName() {
+            return "demo-test";
+        }
+
+        @Bean
+        public Mongo mongo() {
+            return new Fongo("something").getMongo();
```

Comment:
I would use a more meaningful string. I think this is the Fongo DB instance, so something like the class name, or at least "instance", or "defaultInstance" would be better.